### PR TITLE
Add sample snippets to Javadoc

### PR DESCRIPTION
**Pull request by mtopolnik**
Also remove the `mapToOutputFn` parameter from batch `aggregateN` methods.

Removed from `BatchStage`:

- `aggregate2(aggrOp0, stage1, aggrOp1, mapToOutputFn)`
- `aggregate3(aggrOp0, stage1, aggrOp1, stage2, aggrOp2, mapToOutputFn)`

Removed from `BatchStageWithKey`:

- `aggregate(aggrOp, mapToOutputFn)`
- `aggregate2(stage1, aggrOp, mapToOutputFn)`
- `aggregate2(aggrOp0, stage1, aggrOp1, mapToOutputFn)`
- `aggregate3(stage1, stage2, aggrOp, mapToOutputFn)`
- `aggregate3(aggrOp0, stage1, aggrOp1, stage2, aggrOp2, mapToOutputFn)`

Removed `GroupAggregateBuilder.build(mapToOutputFn)`

### CODE REVIEW

**Comment by cangencer** [comment]
verify

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/GeneralStage.java [hunk 1]
```diff
@@ -123,7 +145,16 @@
...
      * <p>
-     * The latency of the async call will add to the latency of items.
+     * The latency of the async call will add to the total latency of the
+     * output.
+     * <p>
+     * Sample usage:
+     * <pre>{@code
+     * stage.mapUsingContextAsync(
+     *     ContextFactory.withCreateFn(jet -> new MyContext(jet)),
+     *     (myContext, item) -> myContext.enrichAsync(item)
```
Comment (vladoschreiner) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/GeneralStage.java]: Function returns completable future. Make this more explicit as this is the tricky part. Such as in https://github.com/hazelcast/hazelcast-jet-code-samples/blob/master/enrichment/src/main/java/Enrichment.java#L246

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/StreamStageWithKey.java [hunk 2]
```diff
@@ -57,6 +57,7 @@
     }
```
Comment (vladoschreiner) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/StreamStageWithKey.java]: No code samples in StreamStageWithKey - is this intentional?

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/GeneralStageWithKey.java [hunk 3]
```diff
@@ -114,11 +133,22 @@
...
+     * this stage for partitioning: all the items with the same key will see
+     * the same context instance (but note that the same instance serves many
+     * keys). One case where this is useful is fetching data from an external
+     * system because you can use a near-cache without duplicating the cached
+     * data.
+     * <p>
+     * Sample usage:
+     * <pre>{@code
+     * items.groupingKey(Item::getDetailId)
+     *      .filterUsingContextAsync(
```
Comment (vladoschreiner) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/GeneralStageWithKey.java]: Code sample for `filterUsingContextAsync` for `filterUsingContext` method

**Pull request merged**